src/article/article.service.ts

```diff
@@ -0,0 +1,128 @@
+  ArticleResponeDto,
+  AuthorDto,
+} from './dto/article-respone.dto';
+
+@Injectable()
+export class ArticleService {
+  constructor(
+    @InjectRepository(Article)
+    private articleRepository: Repository<Article>,
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+  ) {}
+
+  async create(
+    createArticleInputDto: CreateArticleInputDto,
+    userId: number,
+  ): Promise<ArticleResponeDto> {
+    const user = await this.userRepository.findOneBy({ id: userId });
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
+    try {
+      const newArticle = this.articleRepository.create({
+        title: createArticleInputDto.article.title,
+        description: createArticleInputDto.article.description,
+        body: createArticleInputDto.article.body,
+        tagList: createArticleInputDto.article.tagList,
+        authorId: user.id,
+      });
+      const savedArticle = await this.articleRepository.save(newArticle);
+      const userResponse = new AuthorDto({
+        username: user.username,
+        bio: user.bio,
+        image: user.image,
+        following: false,
+      });
+
+      const articleContext = new ArticleDto();
+      articleContext.id = savedArticle.id;
+      articleContext.title = savedArticle.title;
+      articleContext.description = savedArticle.description;
+      articleContext.body = savedArticle.body;
+      articleContext.tagList = savedArticle.tagList;
+      articleContext.author = userResponse;
+      articleContext.createdAt = savedArticle.createdAt.toISOString();
+      articleContext.updatedAt = savedArticle.updatedAt.toISOString();
+
+      const articleRespone = new ArticleResponeDto();
+      articleRespone.article = articleContext;
+      return articleRespone;
+    } catch (err) {
+      if (err instanceof Error) {
+        throw new NotFoundException('Error creating comment: ' + err.message);
+      } else {
+        throw new NotFoundException('An unknown error occurred');
+      }
+    }
+  }
+
+  async update(
+    id: number,
+    updateArticleInputDto: UpdateArticleInputDto,
+    userId: number,
+  ): Promise<ArticleResponeDto> {
+    const user = await this.userRepository.findOneBy({ id: userId });
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
+
+    const updateArticle = await this.articleRepository.findOne({
+      where: {
+        id: id,
+        authorId: userId,
+      },
+    });
+    if (!updateArticle) {
+      throw new NotFoundException(`Article with ID ${id} not found`);
+    }
+
+    try {
+      if (updateArticleInputDto.article.title) {
+        updateArticle.title = updateArticleInputDto.article.title;
+      }
+      if (updateArticleInputDto.article.description) {
+        updateArticle.description = updateArticleInputDto.article.description;
+      }
+      if (updateArticleInputDto.article.body) {
+        updateArticle.body = updateArticleInputDto.article.body;
+      }
```

Comment:
phần này thì ko cần phải check từng field như này xem có thay đổi ko? mà BE sẽ trao đổi với FE/Native dev về cách truyền data trong các trường hợp khác nhau
VD trong TH này là update article thì auto truyền lên hết các tham số, kể cả có thay đổi hay ko, nên API chỉ cần xử lý lưu hết thông tin theo tham số truyền lên là đc